src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts

```diff
@@ -428,25 +428,26 @@ export class DiscussionThread {
   async deleteComment(_id: string): Promise<void> {
     if (this.isLoading[`isDeleting ${_id}`]) return;
 
-    const swrThreadComments = Utils.cloneDeep(this.threadComments);
     this.isLoading[`isDeleting ${_id}`] = true;
-    this.threadComments = this.threadComments.filter(comment => comment._id !== _id);
-    this.threadDictionary = this.arrayToDictionary(this.threadComments);
-
     this.discussionsService.deleteComment(this.discussionId, _id).then((isDeleted: boolean) => {
-      if (!isDeleted) {
-        /* If deletion did not happened, restore original comments */
-        this.threadComments = swrThreadComments;
-      } else {
+      if (isDeleted) {
+        this.threadComments = this.threadComments.filter(comment => comment._id !== _id);
+        this.threadDictionary = this.arrayToDictionary(this.threadComments);
         this.updateDiscussionListStatus(new Date(), this.threadComments.length);
         this.eventAggregator.publish("handleSuccess", "Comment deleted.");
+      } else {
+        Promise.reject("Not deleted.");
```

Comment:
do you have to return this?  what does it do?

Reply:
This should trigger the `catch` in case the `discussionsService.deleteComment` has failed.

Reply:
Just tested again, you would indeed need to return the rejection.
Also throwing an error will also trigger the `.catch()`

Reply:
Better to throw an error object, else the error gets swallowed, because in the `.catch()` block we are using `err.message`.
You would also actually need to add logic to handle `err.message`.
Currently, we only notify user `if (comment._id)` (line 448), and not in other cases